Extensions/Lighting/JsExtension.js

```diff
@@ -0,0 +1,316 @@
+      'Harsimran Virk',
+      'MIT'
+    );
+
+    var lightObstacleBehavior = new gd.BehaviorJsImplementation();
+    lightObstacleBehavior.updateProperty = function (
+      behaviorContent,
+      propertyName,
+      newValue
+    ) {
+      return false;
+    };
+    lightObstacleBehavior.getProperties = function (behaviorContent) {
+      var behaviorProperties = new gd.MapStringPropertyDescriptor();
+
+      return behaviorProperties;
+    };
+    lightObstacleBehavior.initializeContent = function (behaviorContent) {};
+    extension
+      .addBehavior(
+        'LightObstacleBehavior',
+        _('Light Obstacle Behavior'),
+        'LightObstacleBehavior',
+        _('This behavior makes the object an obstacle to light'),
+        '',
+        'CppPlatform/Extensions/topdownmovementicon.png',
+        'LightObstacleBehavior',
+        lightObstacleBehavior,
+        new gd.BehaviorsSharedData()
+      )
+      .setIncludeFile('Extensions/Lighting/lightobstacleruntimebehavior.js');
+
+    var lightObject = new gd.ObjectJsImplementation();
+
+    lightObject.updateProperty = function (
+      objectContent,
+      propertyName,
+      newValue
+    ) {
+      if (propertyName === 'radius') {
+        objectContent.radius = parseFloat(newValue);
+        return true;
+      }
+
+      if (propertyName === 'color') {
+        objectContent.color = newValue;
+        return true;
+      }
+
+      if (propertyName === 'debugMode') {
+        objectContent.debugMode = newValue === '1';
+      }
+
+      return false;
+    };
+
+    lightObject.getProperties = function (objectContent) {
+      var objectProperties = new gd.MapStringPropertyDescriptor();
+
+      objectProperties.set(
+        'radius',
+        new gd.PropertyDescriptor(objectContent.radius.toString())
+          .setType('number')
+          .setLabel(_('Radius'))
+      );
+
+      objectProperties.set(
+        'color',
+        new gd.PropertyDescriptor(objectContent.color).setLabel(_('Color'))
+      );
+
+      objectProperties.set(
+        'debugMode',
+        new gd.PropertyDescriptor(objectContent.debugMode ? 'true' : 'false')
+          .setType('boolean')
+          .setLabel(_('Debug mode'))
+      );
+
+      return objectProperties;
+    };
+    lightObject.setRawJSONContent(
+      JSON.stringify({
+        radius: 50,
+        color: '180,180,180',
+        debugMode: false,
+      })
+    );
+
+    lightObject.updateInitialInstanceProperty = function (
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    lightObject.getInitialInstanceProperties = function (
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'LightObject',
+        _('Light Object for testing'),
```

Comment:
Should it be called _Light_ or _Light Object_? (It's more or less about the consistency of naming, all the other stuff are called object, like _Tiled Sprite Object_, _Text Object_ etc)

Reply:
* In the name *displayed to the user*, we should call it simply "Light", as other objects don't have "Object" in their name (it's implied in the interface, you know that you're dealing with an object).
* For the internal object name, "LightObject" (as you already did) is fine and more explicit. This is especially important because the extension is about "Lighting", which is generic, and then provides to the user multiple "building blocks" which are: an object (LightObject), a behavior for marking objects as casting shadows (LightObstacleBehavior).